cloudcix_primitives/bridgemain.py

```diff
@@ -0,0 +1,436 @@
+        3122: f'Failed to connect to the local host for stop_service payload: ',
+        3123: f'Failed to run stop_service payload on the local host. Payload exited with status ',
+        3124: f'Failed to connect to the local host for delete_files payload: ',
+        3125: f'Failed to run delete_files payload on the local host. Payload exited with status ',
+        3126: f'Failed to connect to the local host for delete_bridge payload: ',
+        3127: f'Failed to run delete_bridge payload on the local host. Payload exited with status ',
+        3128: f'Failed to connect to the local host for reload_services payload: ',
+        3129: f'Failed to run reload_services payload on the local host. Payload exited with status ',
+    }
+
+    def run_host(host, prefix, successful_payloads):
+        fmt = HostErrorFormatter(
+            host,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        # define payloads
+        payloads = {
+            'find_service': f'systemctl status bridgemain_{bridge}.service',
+            'stop_service': f'systemctl stop bridgemain_{bridge}.service && '
+                            f'systemctl disable bridgemain_{bridge}.service',
+            'delete_files': f'rm --force {up_script_path} {down_script_path} {service_file_path}',
+            'reload_services': 'systemctl daemon-reload',
+            'delete_bridge': f'ip link del {bridge}',
+        }
+
+        ret = comms_lsh(payloads['find_service'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f'{prefix + 1}: {messages[prefix + 1]}'), fmt.successful_payloads
+        delete_service = True
+        if ret["payload_code"] != SUCCESS_CODE:
+            delete_service = False
+            fmt.payload_error(ret, f"1101: " + messages[1101]), fmt.successful_payloads
+        fmt.add_successful('find_service', ret)
+
+        if delete_service is True:
+            ret = comms_lsh(payloads['stop_service'])
+            if ret["channel_code"] != CHANNEL_SUCCESS:
+                return False, fmt.channel_error(ret, f'{prefix + 2}: {messages[prefix + 2]}'), fmt.successful_payloads
+            if ret["payload_code"] != SUCCESS_CODE:
+                return False, fmt.payload_error(ret, f'{prefix + 3}: {messages[prefix + 3]}'), fmt.successful_payloads
+            fmt.add_successful('stop_service', ret)
+
+            ret = comms_lsh(payloads['delete_files'])
+            if ret["channel_code"] != CHANNEL_SUCCESS:
+                return False, fmt.channel_error(ret, f'{prefix + 4}: {messages[prefix + 4]}'), fmt.successful_payloads
+            if ret["payload_code"] != SUCCESS_CODE:
+                return False, fmt.payload_error(ret, f'{prefix + 5}: {messages[prefix + 5]}'), fmt.successful_payloads
+            fmt.add_successful('delete_files', ret)
+
+            ret = comms_lsh(payloads['delete_bridge'])
```

Comment:
Is it possible for the bridge service to have been removed but for the bridge to be still on the host? 

Also in build I do see a corresponding command for "adding" bridge. 

When compared with the bridge_kvm primitive, the adding and deleting of bridges is managed in the up.sh and down.sh files the service calls.

In this primitive I think the adding of the bridge is in the up.sh and the deleting of the bridge is called as a payload.

Reply:
Yes, I tested and found that even after removing the service file and restarting systemctl deamon, the bridge is still present, so have to add the last payload to delete the bridge.

Reply:
Oh I got it let me put delete step in down.sh alike bridge_kvm.